server/bin/basket-proxy-server.js

```diff
@@ -59,51 +73,84 @@ function basketRequest(path, method, params) {
       'X-API-Key': API_KEY
     },
     form: params
+  };
+
+  var deferred = p.defer();
+
+  request(config, function (err, httpRequest, body) {
+    if (err) {
+      logger.error('basket.error', err);
+      return deferred.reject(err);
+    }
+
+    if (httpRequest.statusCode !== 200) {
+      // TODO do something here.
+    }
+
+    deferred.resolve(body);
   });
 
-  return req;
+  return deferred.promise;
+}
+
+function forwardBasketResponse(res, path, method, params) {
+  return basketRequest(path, method, params)
+    .then(function (body) {
+      res.type('application/json');
+      res.send(body);
+    }, function (err) {
+      res.status(400).send(err);
```

Comment:
By handling the response from the proxied server (rather than pipe) I assume the basket endpoints will eventually handle the response in the same way? I.e. when we remove the proxy the client logic won't need to be updated because they are also handling errors/responses this way.